flytekit/tools/script_mode.py

```diff
@@ -97,16 +97,14 @@ def tar_strip_file_attributes(tar_info: tarfile.TarInfo) -> tarfile.TarInfo:
 
 
 def fast_register_single_script(
-    wf_entity: WorkflowBase, create_upload_location_fn: typing.Callable
+    source_path: str, module_name: str, create_upload_location_fn: typing.Callable
 ) -> (_data_proxy_pb2.CreateUploadLocationResponse, bytes):
-    _, mod_name, _, script_full_path = extract_task_module(wf_entity)
```

Comment:
We can't get the module name of the imperative workflow by using [inspect.getmodule](https://github.com/flyteorg/flytekit/blob/4f44056bc76e372ac23ce8d388c5be4144ff4b97/flyte) because `inspect.getmodule(<imperative_wf>).__name__` is always equal to `flytekit.core.workflow`

To address this issue, we can use the module name passed in the pyflyte run instead. 
`pyflyte run --remote imperative_wf.py (model_name) wf